keyboards/handwired/yamsek/matrix.c

```diff
@@ -0,0 +1,278 @@
+/* Copyright 2022 Johannes Schneider (@JohSchneider) */
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+/*
+ported from old 2017 project code, while adopting some portions of
+zvecr's mcp23018 based keyboards/viktus/sp111/matrix.c
+*/
+
+#include "quantum.h"
+#include "debounce.h"
+#include "matrix.h"
+#include "mcp23018.h"
+
+#include "yamsek.h"
+
+#include <debug.h>
+
+// Optimize scanning code for speed as a slight mitigation for the port expander
+#pragma GCC push_options
+#pragma GCC optimize("-O3")
+
+// RP2040 with yamsek MCP23017-breakout
+//#define I2C_ADDR 0x32
+// arduino leanardo breadboard
+//#define I2C_ADDR 0b0100000
+#define I2C_ADDR 0x20
+
+
+//#if (MATRIX_ROWS <= 8)
+typedef uint8_t matrix_col_t;
+//#elif (MATRIX_ROWS <= 16)
+//typedef uint16_t matrix_col_t;
+//#endif
+
+/* matrix state(1:on, 0:off) */
+static matrix_col_t matrix[MATRIX_COLS];
+static matrix_col_t raw_matrix[MATRIX_COLS];
+/* matrix is column-major, since the wiring looks like this:
+
+   rows
+
+   R0 ─────│─┬─────│─┬──── ...
+           │ └o o┐ │ └o o┐           -o o- = switch
+           ├───|<┘ ├───|<┘
+           │       │                  -|<- = diode
+   R1 ─────│─┬─────│─┬──── ...
+           │ └o o┐ │ └o o┐
+           ├───|<┘ ├───|<┘
+           │       │
+   .       .       .
+   .       .       .
+           │       │
+           C0      C1      ... columns
+
+  hence the code "selects" column pins by pulling them low,
+  and continues to read all row-pins into a 'matrix_col_t'
+*/
+#if (DIODE_DIRECTION == COL2ROW)
+#    pragma error "DIODE_DIRECTION = COL2ROW is not supported (yet)"
+// TODO: implement like it is done in keyboards/handwired/dactyl/matrix.c?
+#endif
+
```

Comment:
Just remove this — your custom matrix implementation does not actually use the `DIODE_DIRECTION` parameter, and nothing else in QMK needs `DIODE_DIRECTION` to be set.
```suggestion
```